alica_engine/src/engine/syncmodule/SyncModule.cpp

```diff
@@ -38,6 +38,7 @@ SyncModule::SyncModule(ConfigChangeListener& configChangeListener, const TeamMan
 
 SyncModule::~SyncModule()
 {
+    lock_guard<mutex> lock(_lomutex);
```

Comment:
Seems suspicious. I think if you need to lock in the destructor this object can be accessed while being destroyed, wouldn't this be forbidden?

Reply:
I made it for consistency, hope we don't have a situation when we are trying to destroy an object while other thread a inside the method